plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/util/ProcessUtilities.java

```diff
@@ -0,0 +1,266 @@
+   * 
+   * @param commands commands to pass to the {@link ProcessBuilder}
+   * @param workingDir directory to use as the working directory
+   * @param additionalPaths list of additional directories to be appended to the
+   *          PATH environmental variable
+   * @param outputStream output stream to receive process output
+   * @param processReceiver optional, will be given the process right after it
+   *          is started
+   * @return process exit code
+   */
+  public static int launchProcessAndWaitFor(List<String> commands,
+                                            File workingDir,
+                                            final List<String> additionalPaths,
+                                            final OutputStream outputStream,
+                                            IProcessReceiver processReceiver) throws InterruptedException,
+                                                                              IOException {
+    ProcessBuilder pb = new ProcessBuilder(commands);
+    pb.directory(workingDir);
+    pb.redirectErrorStream(true);
+    moveClasspathArgToEnvironmentVariable(commands, pb);
+
+    // if given a non-null, non-empty list of paths, then append to the PATH
+    // environmental variable
+    if (additionalPaths != null && additionalPaths.size() >= 1) {
+      StringBuilder newPathEnvVar = new StringBuilder(pb.environment().get("PATH"));
+      // for each additional path, add it- if it isn't already in the path list
+      for (String path : additionalPaths) {
+        // if this additional path isn't already in the new path environment
+        // variable
+        String[] existingPaths = newPathEnvVar.toString()
+                                              .split(java.io.File.pathSeparatorChar + "");
+        boolean pathAlreadyInPATH = false;
+        for (String existingPath : existingPaths) {
+          if (path.equals(existingPath)) {
+            pathAlreadyInPATH = true;
+          }
+        }
+        if (!pathAlreadyInPATH) {
+          // append the new path onto newPathEnvVar
+          newPathEnvVar.append(java.io.File.pathSeparatorChar);
+          newPathEnvVar.append(path);
+        }
+      }
+      // finally, set the environment variable on the process builder
+      pb.environment().put("PATH", newPathEnvVar.toString());
+    }
+
+    Process process = null;
+    Thread t = null;
+    int processExitCode = -1;
+
+    try {
+      process = pb.start();
+      if (processReceiver != null) {
+        processReceiver.setProcess(process);
+      }
+
+      // Local class to read the output of the process and write it to the
+      // output stream
+      class OutputPump implements Runnable {
+
+        Process process;
+
+        OutputPump(Process p) {
+          process = p;
+        }
+
+        public void run() {
+          BufferedReader processReader = 
+              new BufferedReader(new InputStreamReader(process.getInputStream()));
```

Comment:
need to specify encoding for InputStreamReader. For that matter why do we use a reader here at all instead of just a stream?